src/ServerRequest.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Legionth\React\Http;
+
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\StreamInterface;
+use RingCentral\Psr7\Request;
+
+class ServerRequest extends Request implements ServerRequestInterface
+{
+    private $attributes = array();
+
+    private $serverParams;
+    private $fileParams;
+    private $cookies;
+    private $queryParams;
+    private $parsedBody;
+
+    public function __construct(
+        $method = null,
+        $uri = '',
+        array $headers = array(),
+        StreamInterface $body = null,
+        $protocol = '1.1',
+        array $serverParams = array(),
+        array $fileParams = array(),
+        array $cookies = array(),
+        array $queryParams = array(),
+        $parsedBody = null
```

Comment:
Not sure how useful these parameters are here, as they're not used in the code? Perhaps simplify this and only test the `with*()` methods?